spring-boot-3-integrationtests/src/test/java/org/axonframework/integrationtests/eventsourcing/eventstore/EmbeddedEventStoreTest.java

```diff
@@ -0,0 +1,516 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.integrationtests.eventsourcing.eventstore;
+
+import org.axonframework.common.AxonThreadFactory;
+import org.axonframework.common.transaction.NoOpTransactionManager;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventhandling.*;
+import org.axonframework.eventsourcing.eventstore.EmbeddedEventStore;
+import org.axonframework.eventsourcing.eventstore.EventStorageEngine;
+import org.axonframework.integrationtests.utils.MockException;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.unitofwork.CurrentUnitOfWork;
+import org.axonframework.messaging.unitofwork.DefaultUnitOfWork;
+import org.axonframework.messaging.unitofwork.UnitOfWork;
+import org.axonframework.tracing.TestSpanFactory;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.stream.Collectors.toList;
+import static org.axonframework.integrationtests.utils.EventStoreTestUtils.createEvent;
+import static org.axonframework.integrationtests.utils.EventStoreTestUtils.createEvents;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Test suite validating the {@link EmbeddedEventStore}. Expects end users to make a concrete implementation choosing an
+ * {@link EventStorageEngine} implementation to use during testing.
+ *
+ * @author Rene de Waele
+ * @author Steven van Beelen
+ */
+public abstract class EmbeddedEventStoreTest {
```

Comment:
A similar point as with the DLQ integration test suite applies here.
I think we can come by when we add this dependency so that we do not have to copy it over:
```xml
<dependency>
    <groupId>${project.groupId}</groupId>
    <artifactId>axon-eventsourcing</artifactId>
    <version>${project.version}</version>
    <classifier>tests</classifier>
    <scope>test</scope>
</dependency>
```

Reply:
Yes, I was afraid of dragging in dependencies and persistence.xml files and such, but that can be worked around.